frontend/src/views/Room.jsx

```diff
@@ -105,21 +109,60 @@ function Room() {
     }
   };
 
+  const handleRoleChange = (payload) => {
+    if (payload.eventType === 'INSERT') {
+      const { id, userEmail, permissionId } = payload.new;
+      dispatch(addUpdateParticipant({
+        name: userEmail,
+        role: permissionId === 1 ? ROLES.HOST : ROLES.PRESENTER,
```

Comment:
See my comment above. If the edge function passes a role directly to the Front-End, we don't need to make this operation and are more flexible for when we add more roles in the future.

Reply:
Indeed, I'll change it!